src/os/ui/alert/alertpopup.js

```diff
@@ -127,35 +101,33 @@ export class Controller {
    * @private
    */
   handleAlertEvent_(event) {
-    // deduping for floods of identical messages
-    if (this.lastEvent_) {
-      if (event.getMessage() == this.lastEvent_.getMessage()) {
-        if (!this.dupeDelay_) {
-          this.dupeDelay_ = new Delay(this.resetDupes_, Controller.ALERT_TIMER, this);
-        }
-        this.dupeDelay_.start();
-        this.dupeAlerts_++;
-
-        var limit = event.getLimit();
-        if (this.dupeAlerts_ >= limit && !this.dupeOverflow_) {
-          this.dupeOverflow_ = true;
-
-          // only display the duplicate alert message if the default limit was reached
-          if (limit == DEFAULT_LIMIT) {
-            var dupeEvent = new AlertEvent(
-                'Too many duplicate alerts received! Open the alert window for details.',
-                AlertEventSeverity.WARNING);
-            this.displayAlert_(dupeEvent);
-          }
-        }
-      } else {
-        this.resetDupes_();
+    var display = true;
+
+    if (this.lastEvent_ && event.getMessage() == this.lastEvent_.getMessage()) {
+      // See if a popup for the current alert is showing
+      var msg = event.getMessage();
+      var idx = '';
+      if (msg) {
+        idx = this['alertPopups'].findIndex((popupObj) => popupObj['msg'] == msg);
```

Comment:
It looks like the index is only being used to get the item, so you might want to use `find` instead of `findIndex`, which will return the item directly if found.